jupyter/sum-to-zero/sum_to_zero_evaluation.qmd

```diff
@@ -0,0 +1,1582 @@
+          p[idx] = inv_logit(beta_0 + beta_sex * (i_sex)
+                    + beta_age[i_age] + beta_eth[i_eth] +  beta_edu[i_edu]);
+          p_sample[idx] = p[idx] * sens + (1 - p[idx]) * (1 - spec);
+
+	  // corresponds to likelihood
+          pos_tests[idx] = binomial_rng(tests[idx], p_sample[idx]);
+          idx += 1;
+        }}}}
+```
+
+::: {.callout-note}
+The above set of nested for loops used here to generate the data
+is that same as would be used do to post-stratification the fitted model predictors.
+See section [Coding MRP in Stan](https://mc-stan.org/docs/stan-users-guide/poststratification.html#coding-mrp-in-stan)
+in the Stan User's Guide.
+:::
+
+### Creating Simulated Datasets
+
+The data generating program allows us to create datasets for large and small populations
+and for finer or more coarse-grained sets of categories.
+The larger the number of strata overall, the more observations are needed to get good coverage.
+
+
+##### Instantiate the data generating model.
+
+```{python}
+datagen_model_file = os.path.join('stan', 'gen_binomial_4_preds.stan')
+gen_mod = CmdStanModel(stan_file=datagen_model_file)
+```
+
+##### Specify the number of categories for age, eth, and edu.
+
+```{python}
+gen_data_dict = {
+    'N_eth':3, 'N_edu':5, 'N_age':9, 
+    'baseline': -3.5, 'sens': 0.75, 'spec': 0.9995}
+
+strata = 2 * gen_data_dict['N_age'] * gen_data_dict['N_eth'] * gen_data_dict['N_edu']
+```
+
+###### Specify the total number of observations.
+
+We generate two datasets:  one with a small number of observations, relative to the number of strata,
+and one with sufficient data to provide information on all combinations of demographics.
+
+```{python}
+gen_data = gen_data_dict.copy()
+gen_data['N'] = strata * 17
+
+gen_data_lg = gen_data_dict.copy()
+gen_data_lg['N'] = strata * 200
+
+gen_data_tiny = gen_data_dict.copy()
+gen_data_tiny['N'] = strata * 7
+```
+
+##### Run 1 sampling iteration to get a complete dataset.
+
+```{python}
+sim_data = gen_mod.sample(data=gen_data,
+                          iter_warmup=1, iter_sampling=1, chains=1, seed=45678)
+
+sim_data_lg = gen_mod.sample(data=gen_data_lg,
+                          iter_warmup=1, iter_sampling=1, chains=1, seed=45678)
+
+sim_data_tiny = gen_mod.sample(data=gen_data_tiny,
+                          iter_warmup=1, iter_sampling=1, chains=1, seed=45678)
+```
+
+##### Examine the set of generated data-generating params and resulting dataset.
+
+```{python}
+print(f'Small dataset: N = {gen_data["N"]}, strata = {strata}, expected obs per demographic {gen_data["N"] / strata}')
+print(f'Large dataset: N = {gen_data_lg["N"]}, strata = {strata}, expected obs per demographic {gen_data_lg["N"] / strata}')
+for var, value in sim_data.stan_variables().items():
+    print(var, value[0]) if isinstance(value[0], np.float64) else print(var, value[0][:10])
+```
+
+What is the distribution of the observed number of tests per demographic?
+
```

Comment:
I'd prefer an actual table here with the information and the code secondary just to see how you did it.